crossplane/objects.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+from .parser import parse
+
+
+#
+# Functions
+#
+
+def _init_directive(directive_json):
+    return NGXBlockDirective(**directive_json) if 'block' in directive_json \
+        else NGXDirective(**directive_json)
+
+
+#
+# Classes
+#
+
+class NGXDirective(object):
+    __slots__ = ('directive', 'line', 'args', 'includes')
+
+    def __init__(self, directive='', line=0, args=[], includes=[], **kwargs):
+        self.directive = directive
+        self.line = line
+        self.args = args
+        self.includes = []
+
+    def dict(self):
+        return {
+            slot: getattr(self, slot)
+            for slot in self.__slots__
+        }
+
+
+class NGXBlockDirective(NGXDirective):
+    __slots__ = ('index', 'directive', 'line', 'args', 'block')
+
+    def __init__(self, block=[], **kwargs):
+        super(NGXBlockDirective, self).__init__(**kwargs)
+        self.index = {}
+        self.block = block
+
+        self._setup_block()
+
+    def _setup_block(self):
```

Comment:
if this method is only used in the `__init__` function, why pull the logic out like this?

Reply:
Mostly since I thought it was easier to think of it as a separate step since it is converting JSON into Python objects.  Probably unnecessary personal style.